mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorAllocationHelper.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.master.resourcecluster;
+
+import io.mantisrx.master.resourcecluster.proto.SkuSizeSpec;
+import io.mantisrx.master.resourcecluster.proto.SkuTypeSpec;
+import io.mantisrx.runtime.AllocationConstraints;
+import io.mantisrx.server.master.persistence.IMantisPersistenceProvider;
+import io.mantisrx.server.master.resourcecluster.ClusterID;
+import io.mantisrx.server.master.resourcecluster.ContainerSkuID;
+import io.mantisrx.server.master.resourcecluster.SkuSizeID;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class TaskExecutorAllocationHelper {
+    private final ClusterID clusterID;
+    private final Map<String, String> allocationAttributesWithDefaults;
+    private final IMantisPersistenceProvider storageProvider;
+
+    private Map<ContainerSkuID, SkuTypeSpec> skus = new ConcurrentHashMap<>();
+    private Map<SkuSizeID, SkuSizeSpec> skuSizes = new ConcurrentHashMap<>();
```

Comment:
nit: Assuming this represents the Actor state, we don't need explicit synchronization.